client/flutter/lib/pages/recent_numbers.dart

```diff
@@ -0,0 +1,221 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:who_app/components/page_scaffold/page_scaffold.dart';
+import 'package:who_app/components/recent_numbers_graph.dart';
+import 'package:who_app/generated/l10n.dart';
+import 'package:who_app/api/who_service.dart';
+
+class RecentNumbersPage extends StatefulWidget {
+  @override
+  _RecentNumbersPageState createState() => _RecentNumbersPageState();
+}
+
+class _RecentNumbersPageState extends State<RecentNumbersPage> {
+  var dataTime = 'total';
+  var dataType = 'Cases';
+
+  Map globalStats = Map();
+  DateTime lastUpdated;
+
+  @override
+  void initState() {
+    super.initState();
+    fetchStats();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return PageScaffold(
+      body: <Widget>[
+        CupertinoSliverRefreshControl(
+          builder: (context, refreshIndicatorMode, pulledExtent, b, c) {
+            double topPadding = max(pulledExtent - 25.0, 0.0);
+            switch (refreshIndicatorMode) {
+              case RefreshIndicatorMode.drag:
+                return pulledExtent > 10
+                    ? Padding(
+                        padding: EdgeInsets.only(top: topPadding),
+                        child: Icon(Icons.arrow_downward,
+                            color: CupertinoColors.systemGrey))
+                    : Container();
+                break;
+              case RefreshIndicatorMode.refresh:
+              case RefreshIndicatorMode.armed:
+                return Padding(
+                    padding: EdgeInsets.only(top: topPadding),
+                    child: CupertinoActivityIndicator());
+                break;
+              default:
+                return Container();
+            }
+          },
+          onRefresh: () async {
+            await fetchStats();
+          },
+          refreshIndicatorExtent: 100,
+          refreshTriggerPullDistance: 100,
+        ),
+        SliverList(
+          delegate: SliverChildListDelegate([
+            Container(
+              child: Column(
+                children: <Widget>[
+                  CupertinoSlidingSegmentedControl(
+                    backgroundColor: Color(0xffEFEFEF),
+                    children: _buildSegmentControlChildren(this.dataTime),
+                    groupValue: this.dataTime,
+                    onValueChanged: (value) {
+                      setState(() {
+                        this.dataTime = value;
+                      });
+                    },
+                    padding:
+                        EdgeInsets.symmetric(horizontal: 4.0, vertical: 4.0),
+                    thumbColor: Color(0xffF9F8F7),
+                  ),
+                  Container(height: 16.0),
+                  ConstrainedBox(
+                    child: RecentNumbersGraph(
+                      timeseries: this.globalStats['timeseries'],
+                      timeseriesKey: '${this.dataTime}${this.dataType}',
```

Comment:
this seems pretty fragile.  can we switch these two to enums, and then map from the enum pair to a time series key